src/main/java/com/capgemini/wsb/fitnesstracker/user/internal/UserController.java

```diff
@@ -2,37 +2,95 @@
 
 import com.capgemini.wsb.fitnesstracker.user.api.User;
 import lombok.RequiredArgsConstructor;
-import lombok.extern.java.Log;
-import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.time.LocalDate;
 import java.util.List;
+import java.util.Optional;
+
 
 @RestController
 @RequestMapping("/v1/users")
 @RequiredArgsConstructor
+
 class UserController {
 
     private final UserServiceImpl userService;
 
     private final UserMapper userMapper;
+    private final UserRepository userRepository;
 
     @GetMapping
-    public List<UserDto> getAllUsers() {
-        return userService.findAllUsers()
-                          .stream()
-                          .map(userMapper::toDto)
-                          .toList();
+    public ResponseEntity<List<UserDto>> getAllUsers(@RequestParam(required = false) Integer age) {
+        List<User> users;
+
+        if (age != null) {
+             LocalDate cutoffDate = LocalDate.now().minusYears(age);
+            users = userService.findAllUsers().stream()
+                    .filter(user -> user.getBirthdate().isBefore(cutoffDate))
+                    .toList();
+        } else {
+            users = userService.findAllUsers();
+        }
+
+        List<UserDto> userDtos = users.stream()
+                .map(userMapper::toDto)
+                .toList();
+
+        return ResponseEntity.ok(userDtos);
     }
 
-    @PostMapping
-    public User addUser(@RequestBody UserDto userDto) throws InterruptedException {
+    @PostMapping("/add")
+    public User addUser(@RequestBody User user) throws InterruptedException {
 
         // Demonstracja how to use @RequestBody
-        System.out.println("User with e-mail: " + userDto.email() + "passed to the request");
+        System.out.println("User with e-mail: " + user.getEmail() + "passed to the request");
 
         // TODO: saveUser with Service and return User
-        return null;
+        return userRepository.saveUser(user);
+    }
+
+    @GetMapping("/basic")
+    public List<UserBasicInfoDto> getAllBasicInfo() {
+        return userService.findAllBasicInfo();
+    }
+
+    @GetMapping("/{id}")
+    public Optional<User> getUserById(@PathVariable long id) {
+        return userService.findUserById(id);
+    }
+
+    @DeleteMapping("/delete/id")
+    public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
+        try {
+            userService.deleteUserById(id);
+            return ResponseEntity.noContent().build(); // HTTP 204 No Content on successful deletion
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.notFound().build(); // HTTP 404 Not Found if user not found
+
+        }
+    }
+
+    @PostMapping("/create/{user}")
```

Comment:
Please remove `{user}`, it's should be there. You are passing user as a request body, not as a parameter.